pyNorm/io/pyn_io.py

```diff
@@ -1,3 +1,192 @@
+    except:
+        spec['contin'] = spec_in['ycon']
+        spec['contin_err'] = spec_in['ycon_sig']
+
+    try:
+        spec['contin_order'] = spec_in['order']
+    except:
+        spec['contin_order'] = 0
+
+    spec['contin_coeff'] = np.array([0,0.])
+    #
+    # Construct continuum masks / velocity range
+    #rb_codes saves the mask as a boolean array, need to convert to 0s and 1s before using the
+    #__convert_inorm_mask function
+    masked_area = (spec_in['wc'] == False) #these are the areas masked from rb_codes
+    spec['contin_mask_bits'] = np.zeros(len(spec_in['wc'])) #True = included
+    spec['contin_mask_bits'][masked_area] = 0
+    spec['contin_mask_bits'][~masked_area] = 1
+    spec['mask_cont'] = np.zeros(len(spec_in['wc']))
+    spec['mask_cont'][masked_area] = 0
+    spec['mask_cont'][~masked_area] = 1
+    spec = __convert_inorm_mask(spec)
+    #
+    # Normalized spectrum
+    spec['vnorm'] = spec_in['vel']
+    spec['fnorm'] = spec_in['flux']/spec_in['cont']
+    spec['fnorm_err'] = spec_in['error']/spec_in['cont']
+    spec['fnorm_err_contin'] = 0.
+    spec['fnorm_err_stat'] = 0.
+    #
+    # Na(v) data
+    spec['Nav'] = float('nan')
+    spec['Nav_err'] = float('nan')
+    spec['Nav_sat'] = float('nan')
+    #
+    # Na(v) quantities
+    try:
+        spec['SNR'] = 0.0
+    except:
+        spec['SNR'] = 0.0
+
+    spec['v1'] = spec_in['EWlims'][0]
+    spec['v2'] = spec_in['EWlims'][1]
+    spec['EW'] = spec_in['EW']
+    spec['EW_err'] = spec_in['EWsig']
+    spec['EW_err_stat'] = 0.0
+    spec['EW_err_cont'] = 0.0
+    spec['EW_err_zero'] = 0.0
+    spec['ncol_linearCoG'] = 1.13e17*spec['EW']/(spec['fval']*spec['wavc']**2)
+    spec['ncol_linear2sig'] = 0.0
+    spec['ncol_linear3sig'] = 0.0
+    spec['EW_cumulative'] = np.zeros_like(spec['vel'])
+
+    # Detection flags
+    if spec['EW'] >= 2*spec['EW_err']:
+        print(spec['EW'],spec['EW_err'],type(spec['EW_err']))
+        spec['detection_2sig'] = True
+    else:
+        spec['detection_2sig'] = False
+    if spec['EW'] >= 3*spec['EW_err']:
+        spec['detection_3sig'] = True
+    else:
+        spec['detection_3sig'] = False
+
+    spec['ncol'] = spec_in['N']
+    spec['ncol_err_lo'] = spec_in['Nsig']
+    spec['ncol_err_hi'] = spec_in['Nsig']
+    spec['flag_sat'] = False
+    #
+    spec['va'] = 0.0#spec_in['va'] #average velocity
+    spec['va_err'] = 0.0#spec_in['vaerr']
+    spec['ba'] = 0.0#spec_in['ba'] #velocity width (b value)
+    spec['ba_err'] = 0.0#spec_in['baerr']
+    spec['m3'] = 0.0#spec_in['m3'] #skewness
+    spec['m3_err'] = 0.0#spec_in['m3err']
+    spec['dv90'] = 0.0#spec_in['dv90']
+    spec['v90a'] = 0.0#spec_in['v90a']
+    spec['v90b'] = 0.0#spec_in['v90b']
+
+    # Create an integration limit if not already available
+    if spec['v1'] == spec['v2']:
+        spec['v1'] = -100.
+        spec['v2'] = +100.
+
+    spec = continuum_fit(spec,minord=spec['contin_order']-1,maxord=spec['contin_order']+1)
```

Comment:
Forcing the continuum order to be ±1 around that nominally stored will cause problems if the `contin_order` is already 0 (or perhaps even if it is 1).